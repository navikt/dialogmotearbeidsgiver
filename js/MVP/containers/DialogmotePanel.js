import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Systemtittel } from 'nav-frontend-typografi';
import Icon from '../components/Icon';

const PanelStyled = styled.div`
  padding: 32px;
  background-color: white;
  border-radius: 4px;
`;

const HeaderStyled = styled.header`
  display: flex;
  align-items: center;
  margin-bottom: 32px;
  gap: 16px;
`;

const DialogmotePanel = ({ title, icon, className, children }) => {
  const hasHeader = !!title || !!icon;

  return (
    <PanelStyled className={className}>
      {hasHeader && (
        <HeaderStyled>
          <Icon icon={icon} />
          <Systemtittel>{title}</Systemtittel>
        </HeaderStyled>
      )}
      {children}
    </PanelStyled>
  );
};

DialogmotePanel.propTypes = {
  title: PropTypes.string,
  icon: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node,
};

export default DialogmotePanel;
