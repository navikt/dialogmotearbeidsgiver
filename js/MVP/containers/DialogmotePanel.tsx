import React, { ReactNode } from 'react';
import styled from 'styled-components';
import { Systemtittel } from 'nav-frontend-typografi';

const PanelStyled = styled.div`
  padding: 32px;
  background-color: white;
  border-radius: 4px;
`;

const HeaderStyled = styled.header`
  display: flex;
  align-items: center;
  margin-bottom: 32px;
  gap: 16px;
`;

export interface DialogmotePanelProps {
  title: string;
  icon: ReactNode;
  className?: string;
  children: ReactNode;
}

const DialogmotePanel = ({ title, icon, className, children }: DialogmotePanelProps) => {
  const hasHeader = !!title || !!icon;

  return (
    <PanelStyled className={className}>
      {hasHeader && (
        <HeaderStyled>
          {icon}
          <Systemtittel>{title}</Systemtittel>
        </HeaderStyled>
      )}
      {children}
    </PanelStyled>
  );
};

export default DialogmotePanel;
