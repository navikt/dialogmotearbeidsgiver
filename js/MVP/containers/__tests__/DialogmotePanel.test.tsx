import { IconStub } from '@/MVP/test/stubs/IconStub';
import { render, screen } from '@testing-library/react';
import React from 'react';
import DialogmotePanel from '../DialogmotePanel';

describe('DialogmotePanel', () => {
  test('should render with header', async () => {
    const { container } = render(
      <DialogmotePanel title="Test title" icon={IconStub}>
        <div>Test div</div>
      </DialogmotePanel>
    );

    expect(container).toHaveTextContent('Test title');
  });

  test('should render without header', async () => {
    render(
      <DialogmotePanel>
        <div>Test div</div>
      </DialogmotePanel>
    );

    expect(screen.queryByRole('heading')).not.toBeInTheDocument();
  });
});
