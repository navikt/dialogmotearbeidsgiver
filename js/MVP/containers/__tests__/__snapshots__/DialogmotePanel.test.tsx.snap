// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogmotePanel should render with header 1`] = `
<div>
  <div
    class="sc-gsDKAQ gIzcsR"
  >
    <header
      class="sc-dkPtRN dfjolW"
    >
      <div
        class="sc-bdvvtL jJIaow"
      >
        <img
          alt=""
          src="[object Object]"
        />
      </div>
      <h2
        class="typo-systemtittel"
      >
        Test title
      </h2>
    </header>
    <div>
      Test div
    </div>
  </div>
</div>
`;

exports[`DialogmotePanel should render without header 1`] = `
<div>
  <div
    class="sc-gsDKAQ gIzcsR"
  >
    <div>
      Test div
    </div>
  </div>
</div>
`;
